src/Core/Grid/Definition/Factory/CartGridDefinitionFactory.php

```diff
@@ -0,0 +1,320 @@
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\ActionColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\BulkActionColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\DateTimeColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\DataColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\HighlightedColumn;
+use PrestaShop\PrestaShop\Core\Grid\Filter\Filter;
+use PrestaShop\PrestaShop\Core\Grid\Filter\FilterCollection;
+use PrestaShop\PrestaShop\Core\Hook\HookDispatcherInterface;
+use PrestaShopBundle\Form\Admin\Type\DatalistType;
+use PrestaShopBundle\Form\Admin\Type\DateRangeType;
+use PrestaShopBundle\Form\Admin\Type\SearchAndResetType;
+use PrestaShopBundle\Form\Admin\Type\YesAndNoChoiceType;
+use Symfony\Component\Form\Extension\Core\Type\NumberType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+
+/**
+ * Creates grid definition for Carts grid
+ */
+final class CartGridDefinitionFactory extends AbstractGridDefinitionFactory
+{
+    const GRID_ID = 'cart';
+
+    /**
+     * @var ConfigurationInterface
+     */
+    private $configuration;
+
+    /**
+     * @var string
+     */
+    private $contextDateFormat;
+
+    /**
+     * @var AccessibilityCheckerInterface
+     */
+    private $deleteCartAccessibilityChecker;
+
+    /**
+     * @param HookDispatcherInterface $dispatcher
+     * @param ConfigurationInterface $configuration
+     * @param AccessibilityCheckerInterface $deleteCartAccessibilityChecker
+     * @param string $contextDateFormat
+     */
+    public function __construct(
+        HookDispatcherInterface $dispatcher,
+        ConfigurationInterface $configuration,
+        AccessibilityCheckerInterface $deleteCartAccessibilityChecker,
+        $contextDateFormat
+    ) {
+        parent::__construct($dispatcher);
+
+        $this->configuration = $configuration;
+        $this->contextDateFormat = $contextDateFormat;
+        $this->deleteCartAccessibilityChecker = $deleteCartAccessibilityChecker;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getId()
+    {
+        return self::GRID_ID;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getName()
+    {
+        return $this->trans('Shopping Carts', [], 'Admin.Navigation.Menu');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getColumns()
+    {
+        $columns = (new ColumnCollection())
+            ->add((new BulkActionColumn('cart_bulk_action'))
+                ->setOptions([
+                    'bulk_field' => 'id_cart',
+                ])
+            )
+            ->add((new DataColumn('id_cart'))
+                ->setName($this->trans('ID', [], 'Admin.Global'))
+                ->setOptions([
+                    'field' => 'id_cart',
+                ])
+            )
+            ->add((new DataColumn('status'))
+                ->setName($this->trans('Order ID / Status', [], 'Admin.Orderscustomers.Feature'))
```

Comment:
I don't get this line: both the order ID and the order status will be indicated in the same column?

Reply:
Tested in the UI and see that there was no status - only id:

![image](https://user-images.githubusercontent.com/32385991/73183915-432ef380-4124-11ea-9d52-110f1c762610.png)

Reply:
@tdavidsonas88, thanks! So we should only keep the wording _Order ID_ (and then _Search order ID_, see above), right?

Reply:
right, already implementing it.

Reply:
```suggestion
                ->setName($this->trans('Order ID', [], 'Admin.Orderscustomers.Feature'))
```

Reply:
done

Reply:
Also I think that Raimondas chosen to name field Status / Order Id because of the Abandoned Cart msg:

![image](https://user-images.githubusercontent.com/32385991/73444087-8d9cb400-4360-11ea-9891-3893c7adcacb.png)

In the current code it leaves it as empty Order ID row field.

Reply:
@matks mention this in the standup - looks like Order ID and Status are sometimes mixed. Abandoned cart seems to Cart Status; In the current code implementation in this case left Order ID field value empty if cart has no order. Not sure if it is correct. To avoid misunderstandings - separate Status column seems might be better.

Reply:
This situation happens when new order creation process started but not finished - for example no products added to cart.